services/spar/src/Spar/Scim/User.hs

```diff
@@ -304,6 +383,21 @@ mkValidExternalId (Just idp) (Just extid) = do
               Scim.InvalidValue
               (Just $ "Can't construct a subject ID from externalId: " <> Text.pack err)
 
+primaryEmail' :: forall tag m. (MonadError Scim.ScimError m) => Scim.User tag -> m (Maybe Email)
+primaryEmail' = primaryEmail . Scim.emails
+
+primaryEmail :: forall m. (MonadError Scim.ScimError m) => [ScimEmail.Email] -> m (Maybe Email)
+primaryEmail emails = do
+  for (firstScimEmail emails) $ \scimEmail ->
+    maybe
+      (throwError $ Scim.badRequest Scim.InvalidValue (Just "Not a valid email adress (conflict)"))
+      pure
+      (fromEmailAddress . ScimEmail.unEmailAddress . ScimEmail.value $ scimEmail)
+  where
+    firstScimEmail :: [ScimEmail.Email] -> Maybe ScimEmail.Email
+    firstScimEmail [] = Nothing
+    firstScimEmail (e : es) = Just $ maximumOn1 ScimEmail.primary (e :| es)
```

Comment:
This line makes a lot of implicit assumptions and is a bit hard to read.  What about this?

```suggestion
    firstScimEmail :: [ScimEmail.Email] -> Maybe ScimEmail.Email
    firstScimEmail es = listToMaybe $ (filter (unScimBool . ScimEmail.primary) es) <> es
```

(This is why doctest is so awesome.  You can write completely obfuscated one-liners, as long as you can make their behavior intuitively clear with a handful examples.)

Reply:
Also, please think of better names for `primaryEmail`, `primaryEmail'`: they are not necessarily primary ("first" was better), and the `'` puns nicely on the `primary`, but otherwise is not very helpful.

What about `firstEmailFromUser`, `firstEmailFromEmailsField`?

Reply:
Haha, pun not intendend (unfortunately..). I applied your suggestions